kurtosis-tech/ethereum-package
# feat: add assertoor to additional toolings

Added assertoor testnet testing tool.

There are currently 6 tests included:
* Check for chain stability (finalizing, high voting consensus, no reorgs)
* Check if all clients propose a block
* Validator Lifecycle Test
  Test for the whole validator lifecycle,  includes Deposits, BLSChanges, Exits & Slashings
  This Test takes up to 2 days to run through due to deposit delays & queues. 
* Normal Transaction Test
  Test if normal EOA transactions (type 0/2) are processed by all client pairs (inclusion & submission)
* Blob Transaction Test
  Test if blob transactions (type 3) are processed by all client pairs (inclusion & submission)
* All-Opcodes Transaction Test
  Generates & submits a transaction that triggers every EVM opcode once.
  Checks if the transaction succeeds and no client forks off
